src/components/OcModal.vue

```diff
@@ -106,21 +115,65 @@ export default {
       default: null,
     },
     /**
-     * Text of cancel button
+     * Text of the cancel button
      */
     buttonCancelText: {
       type: String,
       required: false,
       default: "Cancel",
     },
     /**
-     * Text of confirm button
+     * Variation type of the cancel button
+     */
+    buttonCancelVariation: {
+      type: String,
+      required: false,
+      default: "passive",
+      validator: value => {
+        return value.match(/(passive|primary|danger|success|warning)/)
+      },
+    },
+    /**
+     * Appearance of the cancel button
+     */
+    buttonCancelAppearance: {
+      type: String,
+      required: false,
+      default: "outline",
+      validator: value => {
+        return value.match(/(outline|filled|raw)/)
+      },
+    },
+    /**
+     * Text of the confirm button
      */
     buttonConfirmText: {
       type: String,
       required: false,
       default: "Confirm",
     },
+    /**
+     * Variation type of the confirm button
+     */
+    buttonConfirmVariation: {
+      type: String,
+      required: false,
+      default: null,
+      validator: value => {
+        return value.match(/(passive|primary|danger|success|warning)/) || value === null
```

Comment:
Has to be
```suggestion
        return value === null || value.match(/(passive|primary|danger|success|warning)/)
```
otherwise `value.match` will fail hard when `value` evaluates to `null`.